<?xml version="1.0"?>
<launch>
  <arg name="transitions" />
  <arg name="gripperPositionClosed" default="-0.495" />
  <arg name="use_sim" default="false" />

  <arg unless="$(arg use_sim)" name="use_sim_args" value="" />
  <arg     if="$(arg use_sim)" name="use_sim_args" value="--sim" />

  <node name="tobi_picknplace_server" pkg="tobi_picknplace" type="tobi_picknplace_server_node" args="--transitions $(arg transitions) $(arg use_sim_args)" output="screen">
    <param name="gripperPositionClosed" value="$(arg gripperPositionClosed)" />
    <param name="gripperPositionOpen" value="0.298" />
    <param name="graspThroughDistance" value="0.01" />
    <param name="approachDesiredDistance" value="0.10" />
    <param name="approachMinDistance" value="0.05" />
    <param name="liftUpDesiredDistance" value="0.02" />
    <param name="liftUpMinDistance" value="0.005" />
    <param name="retreatDesiredDistance" value="0.07" />
    <param name="retreatMinDistance" value="0.03" />
    <param name="frameOriginArm" value="katana_base_link" />
    <param name="frameOriginGripper" value="katana_gripper_tool_frame" />
    <param name="anglePitchInc" value="0.3" /><!-- pi/8 -->
    <param name="anglePitchMax" value="0.9" /><!-- pi/4 -->
    <param name="angleYawInc" value="0.3" /><!-- pi/8 -->
    <param name="angleYawMax" value="0.9" /><!-- pi/4 -->
    <param name="transInc" value="0.001" />
    <param name="transMax" value="0.0" />
    <param name="placeXInc" value="0.015" />
    <param name="placeYZInc" value="0.04" />
    <param name="placeXMax" value="0.08" />
    <param name="placeYMax" value="0.10" />
    <param name="placeZMax" value="0.17" />
    <param name="heightSkipTop" value="0.005" />
    <param name="heightSkipBottom" value="0.05" />
    <param name="heightInc" value="0.01" />
  </node>

</launch>
